hack/vendor.sh

```diff
@@ -2,9 +2,37 @@
 set -e
 
 cd "$(dirname "$BASH_SOURCE")/.."
-rm -rf vendor/
 source 'hack/.vendor-helpers.sh'
 
+case $# in
+0)
+	rm -rf vendor/
+	;;
+# If user passed arguments to the script
+1)
+	x=$(egrep "^clone [^ ]+ $1" "$0" | sed -E 's,^clone (.*)#.*,\1,')
+	clone $x
```

Comment:
I think this could be slightly simpler, ie:
```bash
eval "$(grep -E "^clone [^ ]+ $1" "$0")"
```
(so that it runs the line from this script as-is, and then we don't need further munging)

Also worth noting, from the `grep` manpage: (http://manpages.debian.org/cgi-bin/man.cgi?query=grep&manpath=Debian+unstable+sid)
>       In addition, the variant programs egrep, fgrep and rgrep are  the  same
>       as  grep -E,  grep -F,  and  grep -r, respectively.  These variants are
>       deprecated, but are provided for backward compatibility.